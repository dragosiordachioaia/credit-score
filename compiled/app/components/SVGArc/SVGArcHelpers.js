"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.drawArc = exports.polarToCartesian = void 0;

var polarToCartesian = function polarToCartesian(centerX, centerY, radius, angleInDegrees) {
  var angleInRadians = (angleInDegrees - 90) * Math.PI / 180.0;
  return {
    x: centerX + radius * Math.cos(angleInRadians),
    y: centerY + radius * Math.sin(angleInRadians)
  };
};

exports.polarToCartesian = polarToCartesian;

var drawArc = function drawArc(_ref) {
  var x = _ref.x,
      y = _ref.y,
      radius = _ref.radius,
      startAngle = _ref.startAngle,
      endAngle = _ref.endAngle;
  var isCompleteCircle = Math.abs(startAngle - endAngle) % 360 == 0;
  var startPoint = polarToCartesian(x, y, radius, endAngle);
  var endPoint = polarToCartesian(x, y, radius, startAngle);
  var largeArcFlag = endAngle - startAngle <= 180 ? 0 : 1;
  return "\n      M ".concat(startPoint.x, " ").concat(startPoint.y, "\n      A ").concat(radius, " ").concat(radius, " 0 ").concat(largeArcFlag, " 0 ").concat(endPoint.x, " ").concat(endPoint.y, "\n      ").concat(isCompleteCircle ? 'Z' : '', "\n  ");
};

exports.drawArc = drawArc;