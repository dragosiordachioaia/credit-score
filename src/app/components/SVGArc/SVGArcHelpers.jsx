export const polarToCartesian = (centerX, centerY, radius, angleInDegrees) => {
    const angleInRadians = (angleInDegrees - 90) * Math.PI / 180.0;
    return {
        x: centerX + (radius * Math.cos(angleInRadians)),
        y: centerY + (radius * Math.sin(angleInRadians))
    };
};

export const drawArc = ({x, y, radius, startAngle, endAngle}) => {
  let isCompleteCircle = Math.abs(startAngle - endAngle) % 360 == 0;
  const startPoint = polarToCartesian(x, y, radius, endAngle);
  const endPoint = polarToCartesian(x, y, radius, startAngle);
  let largeArcFlag = endAngle - startAngle <= 180 ? 0 : 1;

  return `
      M ${startPoint.x} ${startPoint.y}
      A ${radius} ${radius} 0 ${largeArcFlag} 0 ${endPoint.x} ${endPoint.y}
      ${isCompleteCircle ? 'Z' : ''}
  `;
}
