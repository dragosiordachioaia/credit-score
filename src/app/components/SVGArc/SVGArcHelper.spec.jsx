import React from 'react';
import renderer from 'react-test-renderer';

import * as helperMethods from './SVGArcHelpers';

let baseParams = {
  x: 0,
  y: 0,
  radius: 100,
  startAngle: 0,
  endAngle: 0,
}

describe('SVGArc helpers', () => {
  it('drawArc works correctly for 0 degree angle', () => {
    let params = Object.assign({}, baseParams, {endAngle: 0});
    const result = helperMethods.drawArc(params);
    expect(result).toMatchSnapshot();
  });
  it('drawArc works correctly for 90 degree angle', () => {
    let params = Object.assign({}, baseParams, {endAngle: 90});
    const result = helperMethods.drawArc(params);
    expect(result).toMatchSnapshot();
  });
  it('drawArc works correctly for 180 degree angle', () => {
    let params = Object.assign({}, baseParams, {endAngle: 180});
    const result = helperMethods.drawArc(params);
    expect(result).toMatchSnapshot();
  });
  it('drawArc works correctly for 270 degree angle', () => {
    let params = Object.assign({}, baseParams, {endAngle: 270});
    const result = helperMethods.drawArc(params);
    expect(result).toMatchSnapshot();
  });
  it('drawArc works correctly for 360 degree angle', () => {
    let params = Object.assign({}, baseParams, {endAngle: 360});
    const result = helperMethods.drawArc(params);
    expect(result).toMatchSnapshot();
  });
});
