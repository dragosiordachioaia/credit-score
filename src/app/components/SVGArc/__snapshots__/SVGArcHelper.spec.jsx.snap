// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVGArc helpers drawArc works correctly for 0 degree angle 1`] = `
"
      M 6.123233995736766e-15 -100
      A 100 100 0 0 0 6.123233995736766e-15 -100
      Z
  "
`;

exports[`SVGArc helpers drawArc works correctly for 90 degree angle 1`] = `
"
      M 100 0
      A 100 100 0 0 0 6.123233995736766e-15 -100
      
  "
`;

exports[`SVGArc helpers drawArc works correctly for 180 degree angle 1`] = `
"
      M 6.123233995736766e-15 100
      A 100 100 0 0 0 6.123233995736766e-15 -100
      
  "
`;

exports[`SVGArc helpers drawArc works correctly for 270 degree angle 1`] = `
"
      M -100 1.2246467991473532e-14
      A 100 100 0 1 0 6.123233995736766e-15 -100
      
  "
`;

exports[`SVGArc helpers drawArc works correctly for 360 degree angle 1`] = `
"
      M -1.8369701987210297e-14 -100
      A 100 100 0 1 0 6.123233995736766e-15 -100
      Z
  "
`;
